fe/fe-core/src/main/java/org/apache/doris/qe/MasterOpExecutor.java

```diff
@@ -127,19 +129,28 @@ private void forward() throws Exception {
 
         boolean isReturnToPool = false;
         try {
-            result = client.forward(params);
+            client.forward(params);
             isReturnToPool = true;
         } catch (TTransportException e) {
+            // wrap the raw exception.
+            Exception exception = new ForwardToMasterException(
+                    String.format("Forward statement %s to Master %s failed", ctx.getStmtId(),
+                            thriftAddress), e);
+
             boolean ok = ClientPool.frontendPool.reopen(client, thriftTimeoutMs);
             if (!ok) {
-                throw e;
+                throw exception;
             }
             if (shouldNotRetry || e.getType() == TTransportException.TIMED_OUT) {
-                throw e;
+                throw exception;
             } else {
                 LOG.warn("Forward statement " + ctx.getStmtId() + " to Master " + thriftAddress + " twice", e);
-                result = client.forward(params);
-                isReturnToPool = true;
+                try {
+                    client.forward(params);
+                    isReturnToPool = true;
+                } catch (TException ex) {
```

Comment:
maybe use `e` so that the format is consistent with all exception?

Reply:
That may result in a naming confict haha.